operator/pkg/name/name.go

```diff
@@ -80,37 +78,43 @@ var (
 		CitadelComponentName,
 		CNIComponentName,
 	}
-	DeprecatedNames = []ComponentName{
-		InjectorComponentName,
-		SidecarInjectorComponentName,
-		NodeAgentComponentName,
-	}
-	AllLegacyAddonComponentNames = []ComponentName{
-		PrometheusComponentName,
-		KialiComponentName,
-		GrafanaComponentName,
-		TracingComponentName,
-		CoreDNSComponentName,
-	}
-	allComponentNamesMap         = make(map[ComponentName]bool)
-	deprecatedComponentNamesMap  = make(map[ComponentName]bool)
-	LegacyAddonComponentNamesMap = make(map[ComponentName]bool)
-	LegacyAddonComponentPathMap  = make(map[string]string)
+	allComponentNamesMap        = make(map[ComponentName]bool)
+	DeprecatedComponentNamesMap = make(map[ComponentName]bool)
+	AddonComponentNamesMap      = make(map[ComponentName]bool)
```

Comment:
these two still needs some explanation - why are they here and how are they used?

Reply:
done